### PR TITLE
[IA-2585] add retry to app install

### PR DESCRIPTION
**Pull request by jdcanas**
This PR is in response to perf issues. Not much I can do to investigate with db error message and no leo log (which is blown away after perf deploy I think, which we've been doing a lot)

We hit app install and app descriptor retrieval with the hard hammer of retrying everything. We've encountered a lot of instability around helm install in the past, so I think its justified. Unfortunately, I don't know if there's anyway to hone what is retryable from helm install since the go client doesn't provide distinct exceptions.

WB-libs PR this depends on https://github.com/broadinstitute/workbench-libs/pull/576

### CODE REVIEW

File: project/Dependencies.scala
Comment (jdcanas) [inline project/Dependencies.scala]: TODO

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDescriptorDAO.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDescriptorDAO.scala:31]: +1 on this, `http4s` already retries internally I think.

Comment (Qi77Qi) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDescriptorDAO.scala:31]: we set up retry in [here](https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/http/Boot.scala#L353) for all calls through http4s

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: I think this `retry` is in the wrong place. The `installGalaxy` / `installCustomApp` is itself retrying failures, and polling up to 20 minutes until the app becomes running. (In fact [open PR](https://github.com/DataBiosphere/leonardo/pull/2041) makes this process interruptible since we were already retrying retries.)

Can we add the retry with `retryAllConfig` _specifically_ for the `helmClient.installChart` calls? Just want to be careful about not cascading retries.

**Review by rtitle — CHANGES_REQUESTED** [review]
comment above

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala
Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: Yeah, I noticed this morning too. Moved it

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDescriptorDAO.scala
Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpAppDescriptorDAO.scala:31]: Hmm... interesting, I've seen this fail before even with that http4s config which is why I added this here a while ago

File: http/src/main/resources/reference.conf
Comment (rtitle) [inline http/src/main/resources/reference.conf]: Need to revert this… seems made a `reference.conf.bak` file too

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala]: Posted in slack too :) Not sure we want to retry `appDescriptorDAO.getDescriptor` since the `http4s` client already retries exponentially 5 times which should be good enough?

File: project/Dependencies.scala
Comment (rtitle) [inline project/Dependencies.scala]: Should make non-SNAP before merge

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala
Comment (rtitle) [inline http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:1039]: Should we retry the setup chart and nginx helm installs too?

File: http/src/main/resources/reference.conf
Comment (jdcanas) [reply http/src/main/resources/reference.conf]: oof was testing the welder sed locally and forgot to revert

File: http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala
Comment (jdcanas) [reply http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala:1039]: I thought about it, but I think we've mainly seen it around the galaxy-specific helm install (since we started caching the charts) and I wanted to be conservative on this pass. I don't think I've actually seen it fail in those other places post-caching... I'm open to it if you've seen occurrences

**Review by rtitle — APPROVED** [review]
👍

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
Seeing more random `helm install` failures on perf.. gonna merge this once auto tests pass and do another perf run.

**Comment by rtitle** [comment]
Think there's some dependency issue with a library in this PR (which includes a wb-libs version bump), all tests failed and Leo logs full of this:
```
Exception in thread "pool-1-thread-54" java.lang.AbstractMethodError: Receiver class com.google.api.gax.httpjson.HttpJsonCallContext does not define or inherit an implementation of the resolved method 'abstract com.google.api.gax.retrying.RetrySettings getRetrySettings()' of interface com.google.api.gax.retrying.RetryingContext.
	at com.google.api.gax.retrying.ExponentialRetryAlgorithm.createFirstAttempt(ExponentialRetryAlgorithm.java:90)
	at com.google.api.gax.retrying.RetryAlgorithm.createFirstAttempt(RetryAlgorithm.java:112)
	at com.google.api.gax.retrying.BasicRetryingFuture.<init>(BasicRetryingFuture.java:80)
	at com.google.api.gax.retrying.CallbackChainRetryingFuture.<init>(CallbackChainRetryingFuture.java:63)
	at com.google.api.gax.retrying.ScheduledRetryingExecutor.createFuture(ScheduledRetryingExecutor.java:102)
	at com.google.api.gax.rpc.RetryingCallable.futureCall(RetryingCallable.java:61)
	at com.google.api.gax.rpc.RetryingCallable.futureCall(RetryingCallable.java:41)
	at com.google.api.gax.rpc.UnaryCallable$1.futureCall(UnaryCallable.java:126)
	at com.google.api.gax.rpc.UnaryCallable.futureCall(UnaryCallable.java:87)
	at com.google.api.gax.rpc.UnaryCallable.call(UnaryCallable.java:112)
	at com.google.cloud.compute.v1.DiskClient.insertDisk(DiskClient.java:964)
	at com.google.cloud.compute.v1.DiskClient.insertDisk(DiskClient.java:896)
	at org.broadinstitute.dsde.workbench.google2.GoogleDiskInterpreter.$anonfun$createDisk$1(GoogleDiskInterpreter.scala:37)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:104)
	at cats.effect.internals.IORunLoop$.restartCancelable(IORunLoop.scala:51)
	at cats.effect.internals.IOBracket$BracketStart.run(IOBracket.scala:100)
	at cats.effect.internals.Trampoline.cats$effect$internals$Trampoline$$immediateLoop(Trampoline.scala:67)
	at cats.effect.internals.Trampoline.startLoop(Trampoline.scala:35)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.super$startLoop(TrampolineEC.scala:90)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.$anonfun$startLoop$1(TrampolineEC.scala:90)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18)
	at scala.concurrent.BlockContext$.withBlockContext(BlockContext.scala:94)
	at cats.effect.internals.TrampolineEC$JVMTrampoline.startLoop(TrampolineEC.scala:90)
	at cats.effect.internals.Trampoline.execute(Trampoline.scala:43)
	at cats.effect.internals.TrampolineEC.execute(TrampolineEC.scala:42)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:80)
	at cats.effect.internals.IOBracket$BracketStart.apply(IOBracket.scala:58)
	at cats.effect.internals.IORunLoop$.cats$effect$internals$IORunLoop$$loop(IORunLoop.scala:183)
	at cats.effect.internals.IORunLoop$RestartCallback.signal(IORunLoop.scala:463)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:484)
	at cats.effect.internals.IORunLoop$RestartCallback.apply(IORunLoop.scala:422)
	at cats.effect.internals.IOShift$Tick.run(IOShift.scala:36)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by rtitle** [comment]
Dependency issue should be fixed now I think, see https://github.com/broadinstitute/workbench-libs/pull/598

**Pull request merged**
